homeassistant/components/sun/entity.py

```diff
@@ -120,20 +117,19 @@ class Sun(Entity):
     def __init__(self, hass: HomeAssistant) -> None:
         """Initialize the sun."""
         self.hass = hass
+        self.unique_id = ENTITY_ID
```

Comment:
We can't do this. `sun.sun` is hardcoded in many places; we don't want to allow changing it for that reason.

This is also the reason why it isn't an entity component platform. Could you explain / motivate the reason behind this PR in the PR description please?

../Frenck

Reply:
Right, I think it would be quite ok to remove the unique id.

I thought I linked the relevant PR already but apparently not. I have updated the PR description with the linked PR which should provide the context (only allow loading by entity components).

Reply:
👍